From 32c2c9caae5672c1b29d8458a74bbc7ab0e2720e Mon Sep 17 00:00:00 2001
From: Marti Bolivar <mbolivar@leaflabs.com>
Date: Thu, 24 May 2012 17:16:23 -0400
Subject: [PATCH] GPS_PATH_MAX needs to be longer to handle Linux persistent
 device names.

The current value of GPS_PATH_MAX works for /dev/ttyfoo, but is
slightly too small to allow using persistent dev names. For example,
an ND-100S also shows up on my machine as

/dev/serial/by-id/usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0

which requires GPS_PATH_MAX >= 82.

Signed-off-by: Eric S. Raymond <esr@thyrsus.com>
---
 gps.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gps.h b/gps.h
index 6a4920c..0f4d69e 100644
--- a/gps.h
+++ b/gps.h
@@ -43,7 +43,7 @@ extern "C" {
 #define MAXTAGLEN	8	/* maximum length of sentence tag name */
 #define MAXCHANNELS	72	/* must be > 12 GPS + 12 GLONASS + 2 WAAS */
 #define GPS_PRNMAX	32	/* above this number are SBAS satellites */
-#define GPS_PATH_MAX	64	/* dev files usually have short names */
+#define GPS_PATH_MAX	128	/* dev files usually have short names */
 #define MAXUSERDEVS	4	/* max devices per user */
 
 /* 
-- 
1.7.10

